src/playlist.js

```diff
@@ -563,6 +564,101 @@ export const isLowestEnabledRendition = (master, media) => {
   }).length === 0);
 };
 
+export const playlistMatch = (a, b) => {
+  // both playlits are null
+  // or only one playlist is non-null
+  // no match
+  if (!a && !b || (!a && b) || (a && !b)) {
+    return false;
+  }
+
+  // playlist objects are the same, match
+  if (a === b) {
+    return true;
+  }
+
+  // first try to use id as it should be the most
+  // accurate
+  if (a.id && b.id && a.id === b.id) {
+    return true;
+  }
+
+  // next try to use reslovedUri as it should be the
+  // second most accurate.
+  if (a.resolvedUri && b.resolvedUri && a.resolvedUri === b.resolvedUri) {
+    return true;
+  }
+
+  // finally try to use uri as it should be accurate
+  // but might miss a few cases for relative uris
+  if (a.uri && b.uri && a.uri === b.uri) {
+    return true;
+  }
+
+  return false;
+};
+
+export const isAudioOnly = (master) => {
+  const AUDIO = master.mediaGroups && master.mediaGroups.AUDIO || {};
+
+  // we are audio only if we have no main playlists but do
+  // have media group playlists.
+  if (!master.playlists || !master.playlists.length) {
+    for (const groupName in AUDIO) {
+      for (const label in AUDIO[groupName]) {
+        const variant = AUDIO[groupName][label];
+
+        if (variant.playlists && variant.playlists.length || variant.uri) {
+          return true;
+        }
+      }
+    }
+
+    // no audio media groups and no playlists, this cannot be audio only
+    return false;
+  }
+
+  // if every playlist has only an audio codec it is audio only
+  for (let i = 0; i < master.playlists.length; i++) {
+    const playlist = master.playlists[i];
+    const CODECS = playlist.attributes && playlist.attributes.CODECS;
+
+    // all codecs are audio, this is an audio playlist.
+    if (CODECS && CODECS.split(',').every((c) => isAudioCodec(c))) {
+      continue;
+    }
+
+    if (AUDIO) {
+      let audioGroupFound = false;
+
+      for (const groupName in AUDIO) {
+        if (audioGroupFound) {
+          break;
+        }
+        for (const label in AUDIO[groupName]) {
+          const variant = AUDIO[groupName][label];
+
+          // playlist is in an audio group it is audio only
+          if (playlistMatch(playlist, variant)) {
+            audioGroupFound = true;
+            break;
+          }
+        }
+      }
+
+      if (audioGroupFound) {
+        continue;
+      }
+    }
+
+    // if we make it here this playlist isn't audio and we
+    // are not audio only
+    return false;
+  }
```

Comment:
If we can, since this ends up in a nesting of three `for` loops, it may be better to move this to a separate function to check if there's an audio playlist match for a given playlist.

Reply:
added as `someAudioVariant` a function similar to `Array.some`.